lib/dry/configurable.rb

```diff
@@ -51,9 +51,13 @@ def setting(name, type = nil, &block)
       store_reader_key(name) if reader_options?
     end
 
-    def configure(&block)
-      settings_values = ProxySettings.new(struct_class, &block).schema
-      @config = struct_class.new(settings_values)
+    def configure
+      yield(null_config)
```

Comment:
This needs to yield the config struct on subsequent calls, otherwise the config will always be blank on calls to `configure`

Reply:
@AMHOL I think is not necessary since `null_config` will always return the same `NullConfig` object, so no matter how many times you call `configure`. I will add more tests to it so to prove it.

Reply:
I thought with improved dry-types we won't have to do much here, to set defaults. @flash-gordon ?

Reply:
@solnic I think defaults on `dry-types` works as expected. Is a matter of what we want `dry-configurable` works. So what I try with this is approach is for cases where the user specifies default for all the key, so in that case, there is no need to call `configure` he can directly call `finalize!` and he will get his config object.

with a simple configuration, there is no problem
```ruby
class User < Dry::Struct
  attribute :hello, Types::String.default('world')
end

User.new
=> #<User hello="world">
```

the problem comes with a nested configuration 
```ruby
class NestedUser < Dry::Struct
  attribute :nested do
    attribute :name, Types::String.default('world')
  end
end

Nesteduser.new
=> Dry::Struct::Error: [NestedUser.new] :nested is missing in Hash input

NestedUser.new({nested: {}})
=> #<NestedUser nested=#<NestedUser::Nested name="world">>
```

So for simplifying the experience to the user if all the nested keys have defaults and he/she do not want to `configure` he/she should be able to just call `finalize!` and get a config object.

@solnic , @flash-gordon , @timriley , @AMHOL WDYT?

Reply:
FWIW, I've never ever used nested syntax O_o